### DISCUSSION TITLE
how can I implement clicking on one node  to jump and open another node in HTML?

### DISCUSSION THREAD

**Discussion by Seven430** [markmap/markmap#296] [Q&A] [answered]
Thank you for creating such a convenient tool. May I ask, how can I implement clicking on one node in HTML to jump and open another node?

**Comment by gera2ld** [comment]
See #290 for how to get the current node.
You may use `mm.findElement` to get the SVG element of a node, and use `mm.ensureView` to move a node to the viewport.

**Reply by Seven430** [reply]
I've seen this before, but I don't know how to apply it, I'm a newbie, could you give me a little more detail about it ?😊

**Comment by Seven430** [comment]
The code I wrote is as follows:

d3.select("#mindmap").selectAll('foreignObject').on('click', (e, d) => {
                        //console.log('Event:', e);
                        console.log('Data:', d);
                        alert(d.data.state.path);
                        const element = mm.findElement("1.4"); 
                        alert(element);

                        mm.ensureView("1.4");
although"d.data.state.path"=1.4,when i click other element it doesn't work.                 
alert(element)=undefined

**Reply by gera2ld** [reply, marked as answer]
Using markmap-view@0.18:

```ts
d3.select("#mindmap").selectAll('foreignObject').on('click', (e, d) => {
  // If you just want to make the node visible in the viewport
  mm.ensureView(d);

  // If you want to move the element in some way, you need to implement it yourself.
  const result = mm.findElement(d);
  console.log('The node wrapper:', result.g);
});
```
